extensions/objects/models/objects.py

```diff
@@ -779,3 +779,35 @@ class MathExpression(BaseObject):
             'schema': UnicodeString.SCHEMA,
         }]
     }
+
+class Fraction(BaseObject):
+    """Fraction class."""
+
+    description = 'A fraction type'
+    edit_html_filename = 'unicode_string_editor'
+    edit_js_filename = 'FractionEditor'
+    default_value = {
+        'isNegative': False,
+        'wholeNumber': 0,
+        'numerator': 0,
+        'denominator': 1
+    }
+
+    SCHEMA = {
+        'type': 'dict',
+        'properties': [{
+            'name': 'isNegative',
+            'schema': {
+                'type': 'bool'
+            }
+        }, {
+            'name': 'wholeNumber',
+            'schema': NonnegativeInt.SCHEMA
+        }, {
+            'name': 'numerator',
+            'schema': NonnegativeInt.SCHEMA
+        }, {
+            'name': 'denominator',
+            'schema': NonnegativeInt.SCHEMA
```

Comment:
We might want to define a PositiveInt.Schema?

Reply:
Yh I think that makes sense. I added it as a new object type and added some tests (mostly copied from NonnegativeInt).